errorDisplay.js

```diff
@@ -2,109 +2,85 @@
-		gutters = [];
-		editor._codeMirror.clearGutter("errorButton");
-		//Destroy tooltips instance
-		$(".CodeMirror-linenumbers").data("tooltipsy").destroy();
-		//Changes cursor back to default
-		$(".CodeMirror-gutter").removeClass("gutterCursor");
-	}
-	
-	exports.markErrors = markErrors;
-	exports.clearErrors = clearErrors;
-	exports.showWidget = showWidget;
-	exports.showGutter = showGutter;
-	exports.removeGutter = removeGutter;
-	exports.removeWidget = removeWidget;
-});
\ No newline at end of file
+    "use strict";
+
+    var EditorManager  = brackets.getModule("editor/EditorManager"),
+        ExtensionUtils = brackets.getModule("utils/ExtensionUtils"),
+        lineWidgetHTML = require("text!inlineWidget.html"),
+        currentErrorWidget;
+
+    ExtensionUtils.loadStyleSheet(module, "main.less");
+    require("tooltipsy.source");
+
+    function initGutter() {
+        EditorManager.getActiveEditor()._codeMirror.setOption("gutters", ["errors"]);
+    }
+
+    function highlight(line) {
+        if(!line) {
+           return;
+        }
+
+        var editor = EditorManager.getActiveEditor();
+        editor._codeMirror.getDoc().addLineClass(line, "background", "errorHighlight");
+    }
+
+    function removeHighlight(line) {
+        if(!line) {
+           return;
+        }
+
+        var editor = EditorManager.getActiveEditor();
+        editor._codeMirror.getDoc().removeLineClass(line, "background", "errorHighlight");
+    }
+
+    //Function that adds a button on the gutter (on given line nubmer) next to the line numbers
+    function showButton(line){
+        var editor = EditorManager.getActiveEditor();
+        var errorMarker = document.createElement("div");
+        errorMarker.className = "errorButton errorText";
+        errorMarker.innerHTML = "!";
+
+        editor._codeMirror.setGutterMarker(line, "errors", errorMarker);
+
+        //Show tooltips message
+        $(".errors").tooltipsy({content : "Click error icon for details", alignTo: "cursor", offset: [10, -10]});
+    }
+
+    // Function that removes gutter button
+    function removeButton(){
+        var editor = EditorManager.getActiveEditor();
+        editor._codeMirror.clearGutter("errors");
+
+        //Destroy tooltips instance
+        var tooltips = $(".errors").data("tooltipsy");
+        if(tooltips) {
+           tooltips.destroy();
+        }
+    }
+
+    function showDescription(error) {
+        var editor = EditorManager.getActiveEditor();
+        var description = document.createElement("div");
+        description.className = "errorPanel";
+        description.innerHTML = Mustache.render(lineWidgetHTML, {"error": error.message});
+        var options = {coverGutter: false, noHScroll: false, above: false, showIfHidden: false};
+
+        currentErrorWidget = editor._codeMirror.addLineWidget(error.line, description, options);
+    }
+
+    function hideDescription() {
+        if(currentErrorWidget) {
```

Comment:
Invert this, you should return if `!currentErrorWidget` and unindent the rest of the code.